### PR TITLE
feat: add dotenv plugin

### PR DESCRIPTION
**Pull request by salamaashoush**
Port https://github.com/mrsteele/webpack-dotenv

### CODE REVIEW

File: packages/plugin-dotenv/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+import dotenv from 'dotenv-defaults';
+import type { Compiler, RspackPluginInstance, Target } from '@rspack/core';
+import { interpolate, isMainThreadElectron } from './utils';
+
+export interface DotenvPluginOptions {
+  /**
+   * The path to your environment variables.
+   * @default './.env'.
+   */
+  path?: string | undefined;
```
Comment (IWANABETHATGUY) [inline packages/plugin-dotenv/src/index.ts:11]: Since you have already marked the `path` field as optional, the `undefined` looks a little redundant.

Comment (IWANABETHATGUY) [reply packages/plugin-dotenv/src/index.ts:11]: Same for the rest type definition.

**Comment by IWANABETHATGUY** [comment]
The rest part looks good to me. Would you mind having a look @sanyuan0704 about the plugin convention part?

File: packages/plugin-dotenv/src/index.ts (hunk 1 above)
Comment (salamaashoush) [reply packages/plugin-dotenv/src/index.ts]: thanks for spotting it, done

**Review by IWANABETHATGUY — APPROVED** [review]

**Pull request merged**